src/SprykerEco/Zed/AmazonQuicksight/Business/Expander/AnalyticsExpander.php

```diff
@@ -0,0 +1,107 @@
+ */
+
+namespace SprykerEco\Zed\AmazonQuicksight\Business\Expander;
+
+use Generated\Shared\Transfer\AnalyticsCollectionTransfer;
+use Generated\Shared\Transfer\AnalyticsRequestTransfer;
+use Generated\Shared\Transfer\AnalyticsTransfer;
+use Generated\Shared\Transfer\QuicksightUserTransfer;
+use Generated\Shared\Transfer\UserTransfer;
+use SprykerEco\Zed\AmazonQuicksight\Business\ApiClient\AmazonQuicksightApiClientInterface;
+use SprykerEco\Zed\AmazonQuicksight\Persistence\AmazonQuicksightRepositoryInterface;
+use Twig\Environment;
+
+class AnalyticsExpander implements AnalyticsExpanderInterface
+{
+    /**
+     * @var string
+     */
+    protected const TEMPLATE_PATH_QUICKSIGHT_ANALYTICS = '@AmazonQuicksight/_partials/quicksight-analytics.twig';
+
+    /**
+     * @var \SprykerEco\Zed\AmazonQuicksight\Persistence\AmazonQuicksightRepositoryInterface
+     */
+    protected AmazonQuicksightRepositoryInterface $amazonQuicksightRepository;
+
+    /**
+     * @var \SprykerEco\Zed\AmazonQuicksight\Business\ApiClient\AmazonQuicksightApiClientInterface
+     */
+    protected AmazonQuicksightApiClientInterface $amazonQuicksightApiClient;
+
+    /**
+     * @var \Twig\Environment
+     */
+    protected Environment $twigEnvironment;
+
+    /**
+     * @param \SprykerEco\Zed\AmazonQuicksight\Persistence\AmazonQuicksightRepositoryInterface $amazonQuicksightRepository
+     * @param \SprykerEco\Zed\AmazonQuicksight\Business\ApiClient\AmazonQuicksightApiClientInterface $amazonQuicksightApiClient
+     * @param \Twig\Environment $twigEnvironment
+     */
+    public function __construct(
+        AmazonQuicksightRepositoryInterface $amazonQuicksightRepository,
+        AmazonQuicksightApiClientInterface $amazonQuicksightApiClient,
+        Environment $twigEnvironment
+    ) {
+        $this->amazonQuicksightRepository = $amazonQuicksightRepository;
+        $this->amazonQuicksightApiClient = $amazonQuicksightApiClient;
+        $this->twigEnvironment = $twigEnvironment;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AnalyticsRequestTransfer $analyticsRequestTransfer
+     * @param \Generated\Shared\Transfer\AnalyticsCollectionTransfer $analyticsCollectionTransfer
+     *
+     * @return \Generated\Shared\Transfer\AnalyticsCollectionTransfer
+     */
+    public function expandAnalyticsCollectionWithQuicksightAnalytics(
+        AnalyticsRequestTransfer $analyticsRequestTransfer,
+        AnalyticsCollectionTransfer $analyticsCollectionTransfer
+    ): AnalyticsCollectionTransfer {
+        $userTransfer = $analyticsRequestTransfer->getUserOrFail();
+        $quicksightUserTransfer = $this->findQuicksightUser($userTransfer);
+
+        if (!$quicksightUserTransfer) {
+            return $analyticsCollectionTransfer;
+        }
+
+        $userTransfer->setQuicksightUser($quicksightUserTransfer);
+
+        $quicksightGenerateEmbedUrlResponseTransfer = $this->amazonQuicksightApiClient->generateEmbedUrlForRegisteredUser(
```

Comment:
Maybe it makes sense to pass the `QuicksightUserTransfer` to the `generateEmbedUrlForRegisteredUser()` method because we don't use any information from the `UserTransfer` there?